src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,49 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate startDateFormatted = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate endDateFormatted = LocalDate.parse(dateTo, FORMATTER);
+        String[] namesAndSalaries = new String[names.length];
```

Comment:
It's a useless variable. You can create StringBuilder here and write directly
```suggestion
        StringBuilder str = new StringBuilder();
         str.append("Report for period ").append(dateFrom)
                 .append(" - ")
                 .append(dateTo)
                 .append(System.lineSeparator());
```